### DISCUSSION TITLE
What does "files" options actually do?

### DISCUSSION THREAD

**Discussion by anotheruser584** [TypeStrong/ts-node#1348] [Q&A]
I have spent like an hour on a dummy project and still could not get an idea of `files` options.
Form docs: `--files Load files, include and exclude from tsconfig.json on startup`. 
Sorry for stupid question, but I haven't found a solid info on how `ts-node` is working...
Does `ts-node` do in-memory compilation of TypeScript files? Or is `ts-node` using some "temp" folder in `node_modules`? Depending on the mechanism of how `ts-node` is working the explanation above (` Load files, include and exclude from tsconfig.json on startup`) my have different meaning... "Load files..." load where? What is the real-world use case of this options?
Also docs says that `TypeScript Node does not use files, include or exclude, by default.`. 
The `include` and `exclude`  are TypeScript options and it is pretty clear to me. But how are these options affection the `files` options?
So by toggling `files` options I can "load" or "don't load" these files... But still I don't see any difference with this option on/off.
Please help me to figure out the purpose of this option

**Comment by cspotcode** [comment]
The docs for "--files" aren't great.  Like all things open-source, given limited time I've had to prioritize other things.  We would accept a pull request that improves the documentation.

In the meantime, this explanation I posted elsewhere should help at least a little.  https://github.com/TypeStrong/ts-node/discussions/1185#discussioncomment-238837

In short: `ts-node --files` behaves like `tsc`; `ts-node` without `--files` does a performance optimization.

To see the differences, try `tsc --showConfig`, `ts-node --files --showConfig`, and `ts-node --showConfig`.  In the output for all 3, take a close look at the `"files"`, `"include"`, and `"exclude"` arrays.

Again, we will accept a PR improving documentation because I agree that it should be improved.

**Reply by anotheruser584** [reply]
Thanks for the reply and sorry for the late response - I've got sick.
Looking into config shed some light indeed.
If you don't mind I would like to clarify this statement:

> ts-node without --files does a performance optimization.

While testing locally the `files` flag I didn't spot any performance difference. But let's say I'm running e2e tests in NodeJS on something like Azure DevOps or Jenkins or whatever.
I'm running 700+ tests (that are 400+ files) in parallel in Chrome which means the process is already pretty expensive resource vise.
The resource of the virtual machine is limited compared to the local setup because... well, because of money.
Does it mean it would a good idea to turn the `files` flag **off** while running code in the cloud? And if "yes" are there any cons I should be aware of when turning the `files` option off?
p.s. I have found this article that bumped up performance drastically https://medium.com/aspecto/ts-node-ram-consumption-12c257e09e13 so I would like to know what else am I missing.

**Reply by cspotcode** [reply]
Check out our performance recommendations here: https://typestrong.org/ts-node/docs/performance
